.Lib9c.Tests/Action/ReregisterTest.cs

```diff
@@ -0,0 +1,359 @@
+            var orderDigest = order.Digest(avatarState, _tableSheets.CostumeStatSheet);
+            shopState.Add(orderDigest, requiredBlockIndex);
+            orderDigestList.Add(orderDigest);
+
+            Assert.Equal(inventoryCount, avatarState.inventory.Items.Count);
+            Assert.Equal(expectedCount, avatarState.inventory.Items.Sum(i => i.count));
+
+            Assert.Single(shopState.OrderDigestList);
+            Assert.Single(orderDigestList.OrderDigestList);
+
+            Assert.Equal(requiredBlockIndex * 2, sellItem.RequiredBlockIndex);
+            Assert.True(avatarState.inventory.TryGetTradableItems(itemId, requiredBlockIndex * 2, itemCount, out _));
+
+            if (backward)
+            {
+                prevState = prevState.SetState(_avatarAddress, avatarState.Serialize());
+            }
+            else
+            {
+                prevState = prevState
+                    .SetState(_avatarAddress.Derive(LegacyInventoryKey), avatarState.inventory.Serialize())
+                    .SetState(_avatarAddress.Derive(LegacyWorldInformationKey), avatarState.worldInformation.Serialize())
+                    .SetState(_avatarAddress.Derive(LegacyQuestListKey), avatarState.questList.Serialize())
+                    .SetState(_avatarAddress, avatarState.SerializeV2());
+            }
+
+            prevState = prevState
+                .SetState(Addresses.GetItemAddress(itemId), sellItem.Serialize())
+                .SetState(Order.DeriveAddress(order.OrderId), order.Serialize())
+                .SetState(orderDigestList.Address, orderDigestList.Serialize())
+                .SetState(shardedShopAddress, shopState.Serialize());
+
+            var currencyState = prevState.GetGoldCurrency();
+            var price = new FungibleAssetValue(currencyState, ProductPrice, 0);
+            var action = new Reregister
+            {
+                orderId = orderId,
+                reregisterOrderId = reregisterOrderId,
+                tradableId = itemId,
+                sellerAvatarAddress = _avatarAddress,
+                itemSubType = itemSubType,
+                price = price,
+                count = itemCount,
+            };
+            var nextState = action.Execute(new ActionContext
+            {
+                BlockIndex = 101,
+                PreviousStates = prevState,
+                Random = new TestRandom(),
+                Rehearsal = false,
+                Signer = _agentAddress,
+            });
+
+            var reregisterShopAddress = ShardedShopStateV2.DeriveAddress(itemSubType, reregisterOrderId);
+            var nextShopState = new ShardedShopStateV2((Dictionary)nextState.GetState(reregisterShopAddress));
+            Assert.Equal(1, nextShopState.OrderDigestList.Count);
+            Assert.NotEqual(orderId, nextShopState.OrderDigestList.First().OrderId);
+            Assert.Equal(reregisterOrderId, nextShopState.OrderDigestList.First().OrderId);
```

Comment:
It would be good to check the re-registered order information & avatar inventory item information(like expiredBlockIndex & requiredBlockIndex)